### PR TITLE
metrics: Configurable conduit metrics prefix.

### PR DESCRIPTION
**Pull request by winder**
## Summary

Add a configuration to the metrics prefix so that conduit can be different than Indexer.

### CODE REVIEW

File: util/metrics/metrics.go
Comment (shiqizng) [inline util/metrics/metrics.go:14]: typo?

File: util/metrics/metrics.go
Comment (shiqizng) [inline util/metrics/metrics.go:13]: why is deregister() needed?

Comment (winder) [reply util/metrics/metrics.go:13]: It's mainly for tests, which need to make sure these objects are instantiated even if the main init functions are not called. I'll update with a comment.
```suggestion
	// deregister metric objects in case the register function was called more than once.
	// This helps with testing.
	deregister()
```

**Review by Eric-Warehime — APPROVED** [review]
LGTM after you fix typo and resolve merge conflicts.

**Pull request merged**
